controller/upload_file.rst

```diff
@@ -442,10 +442,16 @@ controller.
             public function postLoad(LifecycleEventArgs $args)
             {
                 $entity = $args->getEntity();
-
-                $fileName = $entity->getBrochure();
-
-                $entity->setBrochure(new File($this->targetPath.'/'.$fileName));
+                
+                if (!$entity instanceof Product) {
+                    return;
+                }
+
+                if ($entity->getBrochure()) {
+                    $fileName = $entity->getBrochure();
```

Comment:
Don't really understand this movement, the first if check if the entity passed is the right instance, the second if is launched only if the first pass ... 

If i move the second into the first, that change the whole condition because its suggest that my entity has file (who's not always the case) ... 

Or maybe, you suggest something like that : 

```php
        if (!$entity instanceof Product) {
            return;
        } elseif ($entity->getBrochure()) {
            $fileName = $entity->getBrochure();

            $entity->setBrochure(new File($this->targetPath.'/'.$fileName));
        }
```

Not sure about the movement in fact ...